pkg/restore/actions/pod_volume_restore_action.go

```diff
@@ -112,88 +112,125 @@ func (a *PodVolumeRestoreAction) Execute(input *velero.RestoreItemActionExecuteI
 		podVolumeBackups = append(podVolumeBackups, &podVolumeBackupList.Items[i])
 	}
 	volumeSnapshots := podvolume.GetVolumeBackupsForPod(podVolumeBackups, &pod, podFromBackup.Namespace)
```

Comment:
I would propose another solution:
1. At the beginning of this action, remove the restore-wait init-container blindly without checking needsFileSystemRestore
2. The let it go with the existing logics, if fs-backup is involved, the existing logic will add the restore-wait init-container back

This way looks more simpler.